cirq/google/common_serializers.py

```diff
@@ -118,7 +126,24 @@ def _near_mod_2(e, t, atol=1e-8):
                 gate_getter=lambda x: 'virtual_propagates_forward',
             ),
         ],
-    ),
+        can_serialize_predicate=lambda op: isinstance(op, ops.GateOperation)),
+    op_serializer.GateOpSerializer(
+        gate_type=ops.ZPowGate,
+        serialized_gate_id='z',
+        args=[
+            op_serializer.SerializingArg(
+                serialized_name='half_turns',
+                serialized_type=float,
+                gate_getter='exponent',
+            ),
+            op_serializer.SerializingArg(
+                serialized_name='type',
+                serialized_type=str,
+                gate_getter=lambda x: 'physical_z',
```

Comment:
I feel like it'd be conceptually simpler to have just one serializer for `ZPowGate` that creates correct type here, e.g.:
```python
op_serializer.SerializingArg(
    serialized_name='type',
    serialized_type=str,
    op_getter=lambda op: 'physical' if PhysicalZTag() in op.tags else 'virtual_propagates_forward'
)
```
Obviously this would require changing the arg serializers to allow adding "getters" on the op instead of just the gate, but you've already done this here for `GateOpSerializer` and it seems like we might want it for the args as well.

It also looks like `Operation` does not currently define a `tags` property, but I think that'd be useful so users do not have to always check for TaggedOperation before looking at tags (we did a similar thing with the `gate` property, which is defined on `Operation` even though it could be `None`).

Reply:
Done.  Changed gate_getter to op_getter.

I'll consider adding tags to Operation in a separate PR since it is distinct from this change.